examples/image-processing/main.py

```diff
@@ -0,0 +1,212 @@
+import multiprocessing as mp
+import pickle
+import time
+from glob import glob
+from os import path
+from typing import List, Tuple
+
+import click
+import cv2
+from loguru import logger
+from worker import process_images
+
+import zmq
+
+"""
+    ZMQ client-server for parallel image processing(resize).
+    The server(router) create n workers(dealer): 
+        # each worker can ask a job to the server
+        # a job is an image to resize and a path where the resized_image will be saved  
+        # once, the image was resied, the worker can ask a new job 
+        # server will keep sending job to workers until there is no images left or user hit ctl+c 
+    NOTE: opencv, numpy, click, loguru must be installed 
+"""
+
+"""
+    This program has two mode : 
+        # sequential
+            python main.py sequential-processing --path2initial_images /path/to/source/images --path2resized_images /path/to/target/images --image_extension '*.jpg' --size 512 512
+        # parallel 
+            python main.py parallel-processing --path2initial_images /path/to/source/images --path2resized_images /path/to/target/images --image_extension '*.jpg' --nb_workers 8 --size 512 512
+    parallel mode can be 10x times faster than sequential mode 
+"""
+
+
+@click.group(chain=False, invoke_without_command=True)
+@click.pass_context
+def cmd_group(clk: click.Context):
+    subcommand = clk.invoked_subcommand
+    if subcommand is not None:
+        logger.debug(f'{subcommand} was called')
+    else:
+        logger.debug('use --help to see availables subcommands')
+
+
+@cmd_group.command()
+@click.option(
+    '--path2initial_images', help='initial images location', type=click.Path(True)
+)
+@click.option(
+    '--path2resized_images', help='resized images location', type=click.Path(True)
+)
+@click.option(
+    '--image_extension', help='image file extension', default='*.jpg', type=str
+)
+@click.option(
+    '--size', help='new image size', type=click.Tuple([int, int]), default=(512, 512)
+)
+@click.pass_context
+def sequential_processing(
+    clk: click.Context,
+    path2initial_images: click.Path(True),
+    path2resized_images: click.Path(True),
+    image_extension: str,
+    size: Tuple[int, int],
+):
+    image_filepaths: List[str] = sorted(
+        glob(path.join(path2initial_images, image_extension))
+    )
+    nb_images = len(image_filepaths)
+    logger.debug(f'{nb_images:05d} were found at {path2initial_images}')
+    start = time.time()
+    for cursor, path2source_image in enumerate(image_filepaths):
+        bgr_image = cv2.imread(path2source_image)
+        resized_image = cv2.resize(bgr_image, dsize=size)
```

Comment:
If we're just resizing images, [pillow](https://pillow.readthedocs.io) is a far lighter dependency than a full computer vision library like opencv.

Reply:
This new version use pillow for image resizing, hope this example will help the community to have a better understanding on how to combine zeromq and python-multiprocessing for fast image processing tasks.